pennylane_lightning_gpu/src/util/DataBuffer.cu

```diff
@@ -0,0 +1,37 @@
+//#pragma once
+#include "DataBuffer.hpp"
+
+namespace Pennylane::CUDA {
+
+template void DataBuffer<float2, int>::setElements(int &num_indices,
+                                                   float2 *value, int *indices,
+                                                   bool async);
+template void DataBuffer<double2, int>::setElements(long &num_indices,
+                                                    double2 *value,
+                                                    long *indices, bool async);
```

Comment:
I assume we need to use `int` and `long` here, as well as `float2` and `double2`? Can we map to the associated `cuComplex` types instead --- I know they are the same, but I'd rather keep the types to the intended support & use.

Reply:
sure!